test/cqlpy/test_service_level_api.py

```diff
@@ -107,47 +153,51 @@ def test_count_opened_cql_connections(cql):
 # of CQL connections without restarting them.
-# 
 # This test creates a `test_user` and 2 service levels `sl1` and `sl2`.
 # Firstly the user is assigned to `sl1` and his connections is created.
 # Then the test changes user's service level to `sl2` and 
 # `/service_levels/switch_tenants` endpoint is called.
-def test_switch_tenants(cql):
-    user = f"test_user_{unique_name()}"
-    sl1 = f"sl1_{unique_name()}"
-    sl2 = f"sl2_{unique_name()}"
-    shard_count = get_shard_count(cql)
+def test_appending_new_service_level_no_raft(cql):
+    if is_v2(cql):
+        pytest.skip("Test is not applicable for raft clusters.")
 
-    cql.execute(f"CREATE ROLE {user} WITH login = true AND password='{user}' AND superuser = true")
-    cql.execute(f"CREATE SERVICE LEVEL {sl1} WITH shares = 100")
-    cql.execute(f"CREATE SERVICE LEVEL {sl2} WITH shares = 200")
-    cql.execute(f"ATTACH SERVICE LEVEL {sl1} TO {user}")
-    read_barrier(cql)
+    user, sl1, sl2, shard_count = create_unique_user_service_levels_and_get_shards(cql)
 
     try:
         with new_session(cql, user) as user_session:
-            wait_for_clients(cql, user, 3) # 3 from smp=2 + control connection
-            wait_until_all_connections_authenticated(cql)
-            verify_scheduling_group_assignment(cql, user, sl1, shard_count)
-
-            cql.execute(f"DETACH SERVICE LEVEL FROM {user}")
-            cql.execute(f"ATTACH SERVICE LEVEL {sl2} TO {user}")
-            read_barrier(cql)
-
+            wait_for_clients_and_attach_service_level(cql, user, sl1, sl2, shard_count)
             switch_tenants(cql)
-            # Switching tenants may be blocked if a connection is waiting for a request (see 'generic_server::connection::process_until_tenant_switch()').
-            # Execute enough cheap statements, so that connection on each shard will process at one statement and update its tenant.
+            # Switching tenants may be blocked if a connection is waiting for a 
+            # request (see 'generic_server::connection::process_until_tenant_switch()').
+            # Execute enough cheap statements, so that connection on each 
+            # shard will process at one statement and update its tenant.
             for _ in range(100):
                 read_barrier(user_session)
             verify_scheduling_group_assignment(cql, user, sl2, shard_count)
     finally:
-        cql.execute(f"DETACH SERVICE LEVEL FROM {user}")
-        cql.execute(f"DROP ROLE {user}")
-        cql.execute(f"DROP SERVICE LEVEL {sl1}")
-        cql.execute(f"DROP SERVICE LEVEL {sl2}")
+        finalize_service_level_test(cql, user, [sl1, sl2])
+
+
+# Test if the service level is changed for all excisting connections 
+# while using the cache from group0 this test will only work with raft clusters 
+# firt we will create a new service level and attach it then query it to see 
+# taht all connections are updated immidiatly 
+def test_appending_new_service_level_no_raft(cql):
```

Comment:
There are two functions called `test_appending_new_service_level_no_raft` in this file. I suppose that this one should be called e.g. `test_appending_new_service_level_with_raft`?